utilities_common/multi_asic.py

```diff
@@ -0,0 +1,159 @@
+import subprocess
+import json
+import click
+import functools
+from sonic_device_util import get_num_npus
+from sonic_device_util import get_namespaces
+from sonic_device_util import is_multi_npu
+from sonic_device_util import is_port_internal
+from sonic_device_util import is_port_channel_internal
+from sonic_device_util import is_bgp_session_internal
+from sonic_device_util import get_all_namespaces
+
+import swsssdk
+from swsssdk import ConfigDBConnector
+import argparse
+
+DEFAULT_NAMESPACE = ''
+DISPLAY_ALL = 'all'
+DISPLAY_EXTERNAL = 'frontend'
+PORT_CHANNEL_OBJ = 'PORT_CHANNEL'
+PORT_OBJ = 'PORT'
+BGP_NEIGH_OBJ = 'BGP_NEIGH'
+
+class MultiAsic(object):
+
+    def __init__(self, display_option=DISPLAY_ALL, namespace_option=None):
+        self.namespace_option = namespace_option
+        self.display_option = display_option
+        swsssdk.SonicDBConfig.load_sonic_global_db_config()
+        self.current_namespace = None
+
+    def connect_dbs_for_ns(self,namespace=DEFAULT_NAMESPACE):
```

Comment:
I recently opened a PR to add a "sonic-py-common" library in sonic-buildimage to house common Python functions (https://github.com/Azure/sonic-buildimage/pull/5003). I think some of these functions like this one (which are not CLI-specific) can eventually be moved there, in a `multi_asic.py` file. That way they can be shared by more than just sonic-utilities. This file should hold multi-ASIC functions that are specific to the CLI.

Reply:
@jleveque I agree. I will move the common functions 'sonic-py-common' once the PRs are merged.

Reply:
Great! I have created an issue here to track it: https://github.com/Azure/sonic-utilities/issues/1000